genesyscloud/integration_credential/resource_genesyscloud_integration_credential_test.go

```diff
@@ -161,6 +161,74 @@ func TestAccResourceCredential(t *testing.T) {
 	})
 }
 
+// Resource Credential DEVTOOLING-448
+// This tests to make sure that we can successfully create an integration credential for a Genesys Cloud oauth client without providing a client secret
+func TestAccGenesysCloudOAuthResourceCredential(t *testing.T) {
+	var (
+		oAuthResourceID = "test_genesys_oauth_client"
+		oAuthName       = "test_genesys_oauth_client" + uuid.NewString()
+
+		credResourceID = "test_genesys_oauth_integration_cred"
+		credName       = "Terraform Genesys Oauth Credential Test-" + uuid.NewString()
+	)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { util.TestAccPreCheck(t) },
+		ProviderFactories: provider.GetProviderFactories(providerResources, providerDataSources),
+		Steps: []resource.TestStep{
+			{
+				// Create
+				Config: generateGenesysOauthCredentialResource(oAuthResourceID, oAuthName) + " " + generateOAuthIntegrationCredentialResource(credResourceID, credName, oAuthResourceID),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("genesyscloud_integration_credential."+credResourceID, "name", credName),
+					resource.TestCheckResourceAttr("genesyscloud_oauth_client."+oAuthResourceID, "name", oAuthName),
+				),
+			},
+		},
+		CheckDestroy: testVerifyCredentialDestroyed,
+	})
+}
+
+// TestAccGenesysCloudOAuthResourceCredentialWithSecret will check to make sure we can still create Genesys Cloud
+// integration credential by providing the oauth client and id and secret.  This is how we would normally do it.
+func TestAccGenesysCloudOAuthResourceCredentialWithSecret(t *testing.T) {
```

Comment:
This function is going to make sure we can still create a Genesys Cloud integration credential and still be able to set the clientId and clientSecret.